FlaskApp/__init__.py

```diff
@@ -147,10 +147,34 @@ def register():
         return redirect(url_for("home_page"))
     return render_template("register.html", title="Register", form=form)
 
-@app.route("/login")
-def login():
+@app.route('/login', methods=['GET', 'POST'])
+def log_user_in():
+    if current_user.is_authenticated:
+        flash("Logged in")
+        return redirect(url_for('home_page'))
     form = LoginForm()
-    return render_template("login.html", title="Sign In", form=form)
+    if form.validate_on_submit():
+        user = User.query.filter_by(username=form.username.data).first()
+        if user is None or not user.check_password(form.password.data):
```

Comment:
Can you leave comments here explaining what's going on?

Reply:
Also properly cite any sources if you used a significant chunk of their code/logic